app/src/main/java/com/github/onlynotesswent/ui/overview/editnote/Comments.kt

```diff
@@ -8,100 +8,149 @@ import androidx.compose.foundation.layout.Row
 import com.github.onlynotesswent.R
+import com.github.onlynotesswent.model.file.FileViewModel
 import com.github.onlynotesswent.model.note.Note
 import com.github.onlynotesswent.model.note.NoteViewModel
 import com.github.onlynotesswent.model.user.User
 import com.github.onlynotesswent.model.user.UserViewModel
+import com.github.onlynotesswent.ui.common.ThumbnailDynamicPic
 import com.github.onlynotesswent.ui.navigation.NavigationActions
 import com.github.onlynotesswent.ui.navigation.Screen
-import java.text.SimpleDateFormat
-import java.util.Locale
+import com.github.onlynotesswent.ui.theme.Typography
 
+/**
+ * Screen for viewing and editing comments on a note.
+ * @param navigationActions Actions used for navigating between screens.
+ *  @param noteViewModel The ViewModel for notes.
+ *  @param userViewModel The ViewModel for users.
+ *  @param fileViewModel The ViewModel for files.
+ *
+ */
 @Composable
 fun CommentsScreen(
     navigationActions: NavigationActions,
     noteViewModel: NoteViewModel,
-    userViewModel: UserViewModel
+    userViewModel: UserViewModel,
+    fileViewModel: FileViewModel
 ) {
-  val note by noteViewModel.selectedNote.collectAsState()
-  val currentUser by userViewModel.currentUser.collectAsState()
-  var updatedComments by remember { mutableStateOf(note!!.comments) }
-  Scaffold(
-      floatingActionButton = {
-        AddCommentButton(
-            currentUser = currentUser,
-            note = note,
-            updatedComments = updatedComments,
-            onCommentsChange = { updatedComments = it })
-      },
-      modifier = Modifier.testTag("commentsScreen"),
-      topBar = {
-        EditNoteTopBar(
-            title = stringResource(R.string.comments),
-            titleTestTag = "commentsTitle",
-            noteViewModel = noteViewModel,
-            navigationActions = navigationActions,
-            onClick = {
-              val commentsNotEmpty = updatedComments.commentsList.filter { it.content.isNotEmpty() }
-              noteViewModel.updateNote(
-                  note = note!!.copy(comments = Note.CommentCollection(commentsNotEmpty)))
-            })
-      },
-      bottomBar = {
-        EditNoteNavigationMenu(
-            navigationActions = navigationActions,
-            selectedItem = Screen.EDIT_NOTE_COMMENT,
-            onClick = {
-              val commentsNotEmpty = updatedComments.commentsList.filter { it.content.isNotEmpty() }
-              noteViewModel.updateNote(
-                  note!!.copy(comments = Note.CommentCollection(commentsNotEmpty)))
-              noteViewModel.getNoteById(note!!.id)
-            })
-      }) { paddingValues ->
+    val note by noteViewModel.selectedNote.collectAsState()
+    var noteUpdated by remember { mutableStateOf(note) }
+    val currentUser by userViewModel.currentUser.collectAsState()
+    val updatedComments by remember { derivedStateOf {  (noteUpdated!!.comments) }}
+
+    LaunchedEffect(Unit) {
+        while (true) {
```

Comment:
While true always dangerous but I also am not sure how else you would dynamically update the comment section.